Tests/TestUtils/NetworkRequestHelper.swift

```diff
@@ -81,16 +82,13 @@ class NetworkRequestHelper {
     // MARK: - Network response helpers
     /// Sets the `HttpConnection` response connection for a given `NetworkRequest`
     func setResponseFor(networkRequest: NetworkRequest, responseConnection: HttpConnection?) {
-        networkResponses[networkRequest] = responseConnection
+        let testableNetworkRequest = TestableNetworkRequest(from: networkRequest)
+        networkResponses[testableNetworkRequest] = responseConnection
     }
 
-    /// Gets all network responses for `NetworkRequest`s matching the given `NetworkRequest`
-    ///
-    /// See:
-    func getResponsesFor(networkRequest: NetworkRequest) -> [HttpConnection] {
-        return networkResponses
-            .filter { networkRequest.isCustomEqual($0.key) }
-            .map { $0.value }
+    /// Gets all network responses for the given `NetworkRequest`
```

Comment:
nit: Update comment "Get network response for the given NetworkRequest", plus add comment for parameter and return value.